### PR TITLE
lfdemod refactor

### PR DESCRIPTION
**Pull request by marshmellow42**
re-organize lfdemod.c
refactor detectST function some
refactor pskdemod some.
prep for additional graphing ability to graph demods
add miller decoding for lf.  needs to be tied into data commands. later...
prep for data.c re-factor to come.
i don't want this pull getting bigger so i thought i'd submit even though there is more to come.

### CODE REVIEW

File: common/lfdemod.c [hunk 1]
```diff
@@ -426,543 +308,410 @@ uint32_t manchesterEncode2Bytes(uint16_t datain) {
 
 //by marshmellow
 //encode binary data into binary manchester 
-int ManchesterEncode(uint8_t *BitStream, size_t size)
-{
-	size_t modIdx=20000, i=0;
+//NOTE: BitStream must have double the size available in memory to do the swap
+int ManchesterEncode(uint8_t *BitStream, size_t size) {
+	size_t modIdx=size, i=0;
 	if (size>modIdx) return -1;
```
Comment (pwpiwi) [inline common/lfdemod.c]: This will never be true. Is this intended?

Comment (iceman1001) [inline common/lfdemod.c]: Manchester encoding has the effect of doubling a binary array...

Comment (marshmellow42) [inline common/lfdemod.c]: Good catch.  Looking closer there is another problem.  I'll fix this before we merge.

Comment (marshmellow42) [inline common/lfdemod.c]: i patched this issue and made it better than it was.  ultimately later i think i may add a more array size safe function or a feed and destination array with proper size parameters to check to make sure we can't overrun the array.

Comment (iceman1001) [inline common/lfdemod.c]: :) i misread the question,  now I see it.

**Pull request merged**